BnbnavNetClient/Views/MapView.axaml.cs

```diff
@@ -52,20 +59,64 @@ protected override void OnInitialized()
 
             // We need to pan _more_ when scale is smaller:
             MapViewModel.Pan += (_pointerPrevPosition - pointerPos) / MapViewModel.Scale;
+            
+            _pointerVelocities.Add(new Point(
+                _pointerPrevPosition.X - pointerPos.X,
+                _pointerPrevPosition.Y - pointerPos.Y
+            ));
+
+            if (_pointerVelocities.Count > 5)
+                _pointerVelocities.RemoveAt(0);
+
+            var xSum = 0.0;
+            var ySum = 0.0;
+
+            foreach (var i in _pointerVelocities) {
+                xSum += i.X;
+                ySum += i.Y;
+            }
+
+            _viewVelocity = new(
+                xSum / _pointerVelocities.Count,
+                ySum / _pointerVelocities.Count
+            );
```

Comment:
consider using .Average to make intention clear (though this is fine ig)

Reply:
`List<Point>` does not contain a definition for `Average`? Maybe I should keep this?

Reply:
``IEnumerable<T>.Average`` comes from LINQ... you need to ``using System.Linq;``, at the top, and then use ``var xAvg = _pointerVelocities.Average(i => i.X);`` and ``var yAvg = _pointerVelocities.Average(i => i.Y);``, but yeah ig this is fine, and probably faster too

Reply:
Thanks! Resolved.